core/src/OAuth2/OAuth2Client.cs

```diff
@@ -98,7 +98,7 @@ public void AddBodyParameter(string key, string value)
             {
                 if (method == HttpMethod.Post)
                 {
-                    response = await HttpRequest.SendPost(endpointUri, _headers, _bodyParameters, requestContext);
+                    response = await HttpRequest.SendPost(endpointUri, _headers, new FormUrlEncodedContent(_bodyParameters), requestContext);
```

Comment:
The FormUrlEncodedContent ctor will throw a null ref exception i _bodyParameters is null. It might be safer to check if it's null and if so to send an empty dictionary. 

Style: If you are sure it isn't null, consider adding an Assert.IsTrue(_bodayParameters!=null);

Reply:
The _bodyParameters was initialized as an empty dictionary already.
And this change is not strictly necessary now. I'm reverting back to its original form. #ConsiderItDone